docs/DeveloperGuide.adoc

```diff
@@ -137,111 +137,348 @@ image:BetterModelClassDiagram.png[]
+
+image::UndoRedo/UndoRedoStep5c.png[]
 
-image::CommitActivityDiagram.png[]
+[NOTE]
+Not all commands will save states to StackManager. Exit and Help commands will not save states. Undo and Redo commands should only act on commands that update the items or change the view of the items to the user.
+
+The following activity diagram summarises what happens when a user executes a new command:
+
+image::UndoRedo/UndoRedoActivityDiagram.png[]
 
 ==== Design Considerations
 
 ===== Aspect: How undo & redo executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Saves the entire model.
 ** Pros: Easy to implement.
 ** Cons: May have performance issues in terms of memory usage.
 * **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+** Pros: Will use less memory (e.g. for `delete`, just save the item being deleted).
+** Cons: We must ensure that the implementation of each individual command are correct. +
+Hard to do when we are applying stackable search/sort commands.
 
 ===== Aspect: Data structure to support the undo/redo commands
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
+* **Alternative 1 (current choice):** Use a list to store the history of model states.
 ** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
+** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update the filtered list shown to the user and the backend Xpire data.
 * **Alternative 2:** Use `HistoryManager` for undo/redo
 ** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+=== Search
+
+This feature allows users to filter out specific items either by name or by tag(s) through providing the relevant keyword(s). Items which contain any of the keywords will be shown on the view panel. For search by name, partial words can be matched. For search by tag, only exact words will be matched.
+
+This implementation is under `Logic` and `Model` components.
+
+==== Implementation
+
+Below is the UML sequence diagram of an example usage scenario.
+
+image::SearchCommandSequenceDiagram.png[]
```

Comment:
## Typo in the `SearchCommandSequenceDiagram.png` 
### Image of the typo
![image](https://user-images.githubusercontent.com/54106843/67545168-bb070f80-f72a-11e9-981e-715107fe6fe5.png)

### Comments
If this execute statement is passed into the program, I think there would be syntax error.
Please explain why you chose to make the diagram this way. This is not very epic!

### Suggestions
Would be great if this particular mistake could be rectified because the whole diagram is unreadable now!